File: php/features.php
<?php
declare(strict_types=1);

// Freetestapi SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class FreetestapiFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new FreetestapiBaseFeature();
            case "test":
                return new FreetestapiTestFeature();
            default:
                return new FreetestapiBaseFeature();
        }
    }
}

File: py/features.py
# Freetestapi SDK feature factory

from feature.base_feature import FreetestapiBaseFeature
from feature.test_feature import FreetestapiTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FreetestapiBaseFeature(),
        "test": lambda: FreetestapiTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
